ts/Core/Tooltip.ts

```diff
@@ -1788,6 +1800,92 @@ class Tooltip {
         }
     }
 
+    /**
+     * Checks if a point is within horizontal visible plot and axis bounds.
+     *
+     * @private
+     * @param {number} plotX
+     * @param {Highcharts.Series} series
+     * @param {boolean} paneCoordinates
+     * @return {boolean}
+     */
+    public isInsideX(plotX: number, series?: Series, paneCoordinates?: boolean): boolean {
+        const {
+            inverted,
+            plotBox,
+            plotLeft,
+            scrollablePlotBox,
+            scrollingContainer: {
+                scrollLeft
+            } = { scrollLeft: 0 }
+        } = this.chart;
+
+        const axis = (series && (inverted ? series.yAxis : series.xAxis)) || {
+            pos: plotLeft,
+            len: Infinity
+        };
+
+        if (this.followPointer) {
+            paneCoordinates = false;
+        }
+
+        const x = pick(paneCoordinates, true) ? axis.pos + plotX : plotLeft + plotX;
+
+        return (
+            x >= Math.max(
+                scrollLeft + plotLeft,
+                axis.pos
+            ) &&
+            x <= Math.min(
+                scrollLeft + plotLeft + (scrollablePlotBox || plotBox).width,
+                axis.pos + axis.len
+            )
+        );
+    }
+
+    /**
+     * Checks if a point is within vertical visible plot and axis bounds.
+     *
+     * @private
+     * @param {number} plotY
+     * @param {Highcharts.Series} series
+     * @param {boolean} paneCoordinates
+     * @return {boolean}
+     */
+    public isInsideY(plotY: number, series?: Series, paneCoordinates?: boolean): boolean {
```

Comment:
It looks that `isInsideY` and `isInsideX` are almost the same methods so maybe we can create one common method for both X and Y?